setuptools/tests/integration/test_pip_install_sdist.py

```diff
@@ -25,10 +27,10 @@
 
 from .helpers import Archive, run
 
-
 pytestmark = pytest.mark.integration
 
-(LATEST,) = Enum("v", "LATEST")
+
+(LATEST,) = Enum("v", "LATEST")  # type: ignore[misc] # https://github.com/python/mypy/issues/8009#issuecomment-558335186
```

Comment:
The error here is `error: "Enum" object is not iterable  [misc]`, because we're immediately unpacking the `Enum`. As well as 9 `error: Cannot determine type of "LATEST"  [has-type]` below. The `Enum` type is essentially made anonymous, and lost.

I've linked the reasoning given for not working on allowing this anyway from the mypy devs, but I agree with you that within the scope of a module (or a class attribute, as someone else commented on that thread) is a different scenario.

Still, between that, and the lack of static typing usefulness from the functional API, I doubt this will ever be supported.

This can be done with a class API, but Jason preferred the 1-liner as mentioned here: https://github.com/pypa/setuptools/pull/3979#discussion_r1367963610

Reply:
We probably can use a link for a new issue then: https://github.com/python/mypy/issues/16936.

> Still, between that, and the lack of static typing usefulness from the functional API, I doubt this will ever be supported.

Oh, well, it is very useful... if `mypy` can get the type inference right and does not force you to be explicit when creating constants.

Reply:
```suggestion
(LATEST,) = Enum("v", "LATEST")  # type: ignore[misc] # https://github.com/python/mypy/issues/16936
```

Reply:
Personally I'm fine with
```suggestion
v = Enum("v", "LATEST")
LATEST = v.LATEST
```
https://github.com/python/mypy/issues/16936#issuecomment-1959594042
But I'll leave that decision up to you.

Reply:
The way it is should be fine. It is a nice syntax, because it is a one-liner after all...